src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_seckey.c

```diff
@@ -139,3 +139,82 @@ uint64_t AppleCryptoNative_SecKeyGetSimpleKeySizeInBytes(SecKeyRef publicKey)
 
     return SecKeyGetBlockSize(publicKey);
 }
+
+#if defined(TARGET_MACCATALYST) || defined(TARGET_IOS) || defined(TARGET_TVOS)
```

Comment:
Does this need to be conditional?

* No: Great, let's remove the ifs
* Yes, this uses API not on macOS: Sadface.
* Yes, this is alternative implementations of functions already present for macOS: Can we get rid of all of these triple target references and use one or two high-level defines based on these?  (e.g. FULL_MACOS and MOBILE_MACOS, or some names that get more than 3 seconds of thought)

Reply:
It doesn't have to be conditional. It's just not used on macOS yet in this PR which is why I excluded it. I'd be happy to remove the conditionals and add the references to `entrypoints.c`. I prototypes managed implementation that uses both types of the keys  (iOS / CSSM) on macOS but I don't intent to submit it as PR (at least not until the the iOS part is working).

Generally I would prefer to split the iOS / macOS code into different files in the native dylib but I didn't do it because of the preexisting code. I do think it would make sense though, especially with support for iOS keychain which is substantially different from the macOS one. Most of the `#if`s would be removed and apper only once in the CMake script that way.

Reply:
Yeah, I guess for me it's "is this only for iOS" vs "is this only for iOS **today**".  If future PRs (particularly in this release) are going to start calling these functions from macOS, the API shouldn't be conditional.

And if the API is conditional, I agree that I'd prefer the isolation be done at the file level.

Reply:
I guess it is iOS only *today* so I'll just include these new APIs unconditionally.

> And if the API is conditional, I agree that I'd prefer the isolation be done at the file level.

If I get to do the iOS keychain work I'll introduce that pattern.

Reply:
I've opened #51976 to get CI run for the implementation of data keys on macOS. I suspect it will need some tweaking but given how much code I'd move back and forth it would likely make sense to merge it into this PR if I can get the CI to pass.